Hardcode shadersDir to the correct path in Debian in various plugins
 TODO: Put a short summary on the line above and replace this paragraph
 with a longer explanation of this change. Complete the meta-information
 with other relevant fields (see below for details). To make it easier, the
 information below has been extracted from the changelog. Adjust it or drop
 it.
 .
 meshlab (1.3.3) precise; urgency=low
 .
   * new snapshot release
Author: Stefan Schlager <zarquon42@gmail.com>

---
The information above should follow the Patch Tagging Guidelines, please
checkout http://dep.debian.net/deps/dep3/ to learn about the format. Here
are templates for supplementary fields that you might want to add:

Origin: <vendor|upstream|other>, <url of original patch>
Bug: <url in upstream bugtracker>
Bug-Debian: http://bugs.debian.org/<bugnumber>
Bug-Ubuntu: https://launchpad.net/bugs/<bugnumber>
Forwarded: <no|not-needed|url proving that it has been forwarded>
Reviewed-By: <name and email of someone who approved the patch>
Last-Update: <YYYY-MM-DD>

--- meshlab-1.3.3.orig/meshlab/src/plugins_experimental/render_splatpyramid/dialog.cpp
+++ meshlab-1.3.3/meshlab/src/plugins_experimental/render_splatpyramid/dialog.cpp
@@ -123,7 +123,7 @@ void Dialog::changeTexturePath(int i) {
 void Dialog::browseTexturePath(int i) {
   QFileDialog fd(0,"Choose new texture");
 
-  QDir shadersDir = QDir(qApp->applicationDirPath());
+  QDir shadersDir = QDir("/usr/share/meshlab");
 #if defined(Q_OS_WIN)
   if (shadersDir.dirName() == "debug" || shadersDir.dirName() == "release")
 	shadersDir.cdUp();
--- meshlab-1.3.3.orig/meshlab/src/plugins_experimental/render_splatpyramid/splat_pyramid.cc
+++ meshlab-1.3.3/meshlab/src/plugins_experimental/render_splatpyramid/splat_pyramid.cc
@@ -1,3 +1,4 @@
+
 /**
  * Interface independent application class
  * 
@@ -85,7 +86,7 @@ void PyramidPointRendererPlugin::Init(QA
 		objects[i].setRendererType( render_mode );
 
   	/// Sets the path to the meshlab's shaders directory
-  	QDir shadersDir = QDir(qApp->applicationDirPath());
+  	QDir shadersDir = QDir("/usr/share/meshlab");
 #if defined(Q_OS_WIN)
 	if (shadersDir.dirName() == "debug" || shadersDir.dirName() == "release" || shadersDir.dirName() == "plugins"  )
 		shadersDir.cdUp();
--- meshlab-1.3.3.orig/meshlab/src/meshlabplugins/render_gdp/meshrender.cpp
+++ meshlab-1.3.3/meshlab/src/meshlabplugins/render_gdp/meshrender.cpp
@@ -37,7 +37,7 @@ void MeshShaderRenderPlugin::initActionL
 	qaNone->setCheckable(false);
 	actionList << qaNone;*/
 
-	QDir shadersDir = QDir(qApp->applicationDirPath());
+	QDir shadersDir = QDir("/usr/share/meshlab");
 #if defined(Q_OS_WIN)
 	if (shadersDir.dirName() == "debug" || shadersDir.dirName() == "release" || shadersDir.dirName() == "plugins"  )
 		shadersDir.cdUp();
--- meshlab-1.3.3.orig/meshlab/src/meshlabplugins/render_gdp/shaderDialog.cpp
+++ meshlab-1.3.3/meshlab/src/meshlabplugins/render_gdp/shaderDialog.cpp
@@ -329,7 +329,7 @@ void ShaderDialog::changeTexturePath(int
 void ShaderDialog::browseTexturePath(int i) {
 	QFileDialog fd(0,"Choose new texture");
 
-	QDir shadersDir = QDir(qApp->applicationDirPath());
+	QDir shadersDir = QDir("/usr/share/meshlab");
 #if defined(Q_OS_WIN)
 	if (shadersDir.dirName() == "debug" || shadersDir.dirName() == "release")
 		shadersDir.cdUp();
--- meshlab-1.3.3.orig/meshlab/src/meshlabplugins/render_rfx/render_rfx.cpp
+++ meshlab-1.3.3/meshlab/src/meshlabplugins/render_rfx/render_rfx.cpp
@@ -55,7 +55,7 @@ QList<QAction*> RenderRFX::actions()
 
 void RenderRFX::initActionList()
 {
-	QDir shadersDir = PluginManager::getBaseDirPath();
+	QDir shadersDir = QDir("/usr/share/meshlab");
 
 #if defined(Q_OS_WIN)
 	if (shadersDir.dirName() == "debug"   ||
--- meshlab-1.3.3.orig/meshlab/src/plugins_unsupported/render_rm/rmmeshrender.cpp
+++ meshlab-1.3.3/meshlab/src/plugins_unsupported/render_rm/rmmeshrender.cpp
@@ -63,7 +63,7 @@ QList<QAction*> RmMeshShaderRenderPlugin
 
 void RmMeshShaderRenderPlugin::initActionList() {
 
-	QDir shadersDir = QDir(qApp->applicationDirPath());
+	QDir shadersDir = QDir("/usr/share/meshlab");
 #if defined(Q_OS_WIN)
 	if (shadersDir.dirName() == "debug" ||
 	    shadersDir.dirName() == "release" ||
